Build images using image creator.

psladek opened the pull request:
- allows general SLEPOS image

okurz commented on tests/slepos/build_images_creator.pm:
```diff
@@ -0,0 +1,254 @@
+    save_screenshot;
+    #todo: uncheck 32bit for (get_var('VERSION') =~ /^12/) {
+    assert_and_click('image-creator-next1');
+    if (check_screen('import-untrusted-gpg-key')) {
```
please try

```perl
assert_screen [qw(import-untrusted-gpg-key image-creator-image-configuration)];
send_key 'alt-i' if match_has_tag 'import-untrusted-key';
```

This can save the 30s waiting time in case of no untrusted key

okurz commented on tests/slepos/build_images_creator.pm:
```diff
@@ -0,0 +1,254 @@
+        send_key('alt-a');
+        if (check_screen('changed-packages')) {
+            send_key('alt-o');
+        }
```
like above, better use `assert_screen` with multiple tags and `match_has_tag`

okurz commented on tests/slepos/build_images_creator.pm:
```diff
@@ -0,0 +1,254 @@
+    my $images_ref = get_var_array('IMAGE_CREATOR');
+
+    save_screenshot;
+    select_console('x11');
```
that will now by default assert the console '11' to show up, e.g. "generic-desktop". Because you are using "ensure_unlocked_desktop" afterwards I recommend to say `select_console 'x11', await_console => 0;`

okurz commented on tests/slepos/build_images_creator.pm:
```diff
@@ -0,0 +1,254 @@
+
+    #start image creator
+    x11_start_program("xdg-su -c '/sbin/yast2 image-creator'", 3);
+    if (check_screen('root-auth-dialog')) {
```
see above

okurz commented on tests/slepos/build_images_creator.pm:
```diff
@@ -0,0 +1,254 @@
+    save_screenshot;
+
+    #start image creator
+    x11_start_program("xdg-su -c '/sbin/yast2 image-creator'", 3);
```
that is a very short timeout

okurz commented on tests/slepos/register_images.pm:
```diff
@@ -17,13 +17,53 @@ use testapi;
 use lockapi;
 
+my $img_suffix = 'tar.bz2';
+
```
the file itself should not have executable bit set

nadvornik commented on tests/slepos/register_images.pm:
```diff
@@ -17,13 +17,53 @@ use testapi;
+        my $target = get_var("ASSET_$idx");
+        $target =~ s/^\d+-//;                #remove if private assets reference
+        $target =~ s/\Q.$img_suffix\E$//;    #remove suffix
+        @dwl_images = (@dwl_images, $target);
```
There seems to be some code duplication between this and download_images.pm. Maybe you could move the image name processing to a library.

okurz commented on tests/slepos/build_images_creator.pm:
```diff
@@ -0,0 +1,256 @@
+# SUSE's openQA tests
+#
+# Copyright © 2016 SUSE LLC
```
2017?

okurz commented on tests/slepos/build_images_creator.pm:
```diff
@@ -0,0 +1,256 @@
+my %template_done;
+
+sub build_image {
+
```
delete blank line

okurz commented on tests/slepos/build_images_creator.pm:
```diff
@@ -0,0 +1,256 @@
+        $template_done{$template} = 1;
+
+        select_console('x11');
+        wait_still_screen;
```
that does not make sense to me. `select_console` already asserts the "x11" console so wait_still_screen should not be needed

okurz commented on tests/slepos/build_images_creator.pm:
```diff
@@ -0,0 +1,256 @@
+    #add virtual drivers unless already done
+    if (!$template_done{$template}) {
+        select_console('root-console');
+        wait_still_screen;
```
see below

okurz replied:
I read your comment that it's needed but select_console should implicitly call an assert_screen afterwards on the console being ready before typing. Also the `save_screenshot` seems redundant to me. Please check that you have an updated os-autoinst.

okurz commented on tests/slepos/build_images_creator.pm:
```diff
@@ -0,0 +1,256 @@
+    my ($image, $template, $template_idx, $packages_ar) = @_;
+
+    my ($imagebase, $version) = split_image_name($image);
+    bmwqemu::diag("image '$image' will be built using image creator:");
```
please avoid use of `bmwqemu`. `diag` is part of testapi directly since some time. The ":" does not make sense to me as there is nothing coming afterwards

okurz commented on tests/slepos/build_images_creator.pm:
```diff
@@ -0,0 +1,256 @@
+
+    #upload image
+    select_console('root-console');
+    wait_still_screen;
```
should not be needed

okurz commented on tests/slepos/build_images_creator.pm:
```diff
@@ -0,0 +1,256 @@
+
+    select_console('x11', await_console => 0);
+    ensure_unlocked_desktop;
+
```
delete blank line

okurz commented on tests/slepos/build_images_creator.pm:
```diff
@@ -0,0 +1,256 @@
+
+
+sub prepare_template {
+
```
delete blank line

okurz commented on tests/slepos/build_images_creator.pm:
```diff
@@ -0,0 +1,256 @@
+    type_string "cat $image_path/$template/config.xml\n";
+    wait_still_screen;
+    save_screenshot;
+
```
delete blank line

okurz commented on tests/slepos/build_images_creator.pm:
```diff
@@ -0,0 +1,256 @@
+
+
+sub run() {
+
```
delete blank line

okurz commented on tests/slepos/build_images_creator.pm:
```diff
@@ -0,0 +1,256 @@
+    send_key('alt-l');                          #close
+    select_console('root-console');
+    wait_still_screen;
+
```
delete blank line

okurz commented on tests/slepos/build_images_creator.pm:
```diff
@@ -0,0 +1,256 @@
+}
+
+
+
```
delete two blank lines

psladek commented:
- changed date, removed empty lines to accomodate local style
- keeping 'wait_still_screen' cmds for safety, since at least in the first occurence it is really needed, otherwise it results into 'rep' being written on the commandline instead of 'grep' in the next step.

asmorodskyi commented:
even taking into account your comment still agree with @okurz here, we should not have wait_still screen after each select_console. Need just fix select_console behavior so it pass control only after console is really selected. But for sure not in terms of this PR

okurz commented:
> Need just fix select_console behavior so it pass control only after console is really selected. But for sure not in terms of this PR

select_console *is already* implemented in a way to only return after a corresponding needle has been found.

@psladek make sure your local repository is up to date. In case you have problem please reference a URL to a failing test case. Then we can take a look and help.

asmorodskyi commented:
@psladek  what your plans regarding this PR ?
